cpp/ql/src/Likely Bugs/Format/NonConstantFormat.ql

```diff
@@ -79,38 +84,24 @@ predicate isNonConst(DataFlow::Node node, boolean isIndirect) {
       )
     )
     or
-    exists(Parameter p | p = e.(VariableAccess).getTarget() |
-      p.getFunction().getName() = "main" and p.getType() instanceof PointerType
-    )
-    or
-    e instanceof CrementOperation
-    or
-    e instanceof AddressOfExpr
-    or
-    e instanceof ReferenceToExpr
-    or
-    e instanceof AssignPointerAddExpr
-    or
-    e instanceof AssignPointerSubExpr
-    or
-    e instanceof PointerArithmeticOperation
-    or
-    e instanceof FieldAccess
-    or
-    e instanceof PointerDereferenceExpr
-    or
-    e instanceof AddressOfExpr
-    or
-    e instanceof ExprCall
-    or
-    e instanceof NewArrayExpr
-    or
     exists(Variable v | v = e.(VariableAccess).getTarget() |
       v.getType().(ArrayType).getBaseType() instanceof CharType and
       exists(AssignExpr ae |
         ae.getLValue().(ArrayExpr).getArrayBase().(VariableAccess).getTarget() = v
       )
     )
+    or
+    exists(UncalledFunction f, Parameter p | f.getAParameter() = p |
+      p = e.(VariableAccess).getTarget()
+    )
```

Comment:
I guess this covers the `main` arguments case that was removed.  What else do you expect we gain with the more general case here?

Reply:
APIs. Most everything I deal with has no main method, but it has plenty of user endpoints as APIs.

Reply:
Makes perfect sense, thanks.